contrib/qpl-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,362 @@
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_icf_finish.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_inflate_multibinary.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_update_histogram_01.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_update_histogram_04.asm
+                 ${QPL_SRC_DIR}/isal/igzip/rfc1951_lookup.asm
+                 ${QPL_SRC_DIR}/isal/igzip/adler32_sse.asm
+                 ${QPL_SRC_DIR}/isal/igzip/adler32_avx2_4.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_deflate_hash.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_set_long_icf_fg_04.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_set_long_icf_fg_06.asm
+                 ${QPL_SRC_DIR}/isal/igzip/igzip_multibinary.asm
+                 ${QPL_SRC_DIR}/isal/igzip/stdmac.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc_multibinary.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_gzip_refl_by8.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_gzip_refl_by8_02.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_gzip_refl_by16_10.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_ieee_01.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_ieee_02.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_ieee_by4.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_ieee_by16_10.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_iscsi_00.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_iscsi_01.asm
+                 ${QPL_SRC_DIR}/isal/crc/crc32_iscsi_by16_10.asm)
+
+set(DISABLED_WARNINGS_FLAGS /wd4244 /wd4101 /wd4267 /wd4018 /wd4146 /wd4334 /wd4308)
+set(FEATURE_FLAGS -DHAVE_AS_KNOWS_AVX512 -DAS_FEATURE_LEVEL=10)
+
+# Adding ISA-L library target
+add_library(isal OBJECT ${ISAL_C_SRC})
+add_library(isal_asm OBJECT ${ISAL_ASM_SRC})
+
+# Setting external and internal interfaces for ISA-L library
+target_include_directories(isal
+                           PUBLIC $<BUILD_INTERFACE:${QPL_SRC_DIR}/isal/include>
+                           PRIVATE ${QPL_SRC_DIR}/isal/include
+                           PUBLIC ${QPL_SRC_DIR}/isal/igzip)
+
+set_target_properties(isal PROPERTIES
+                      CXX_STANDARD 11
+                      C_STANDARD 99)
+
+target_compile_options(isal PRIVATE
+                       "$<$<C_COMPILER_ID:GNU>:${QPL_LINUX_TOOLCHAIN_REQUIRED_FLAGS}>"
+                       "$<$<C_COMPILER_ID:MSVC>:/WX;${DISABLED_WARNINGS_FLAGS};${QPL_WINDOWS_TOOLCHAIN_REQUIRED_FLAGS}>"
+                       "$<$<CONFIG:Debug>:>"
+                       "$<$<CONFIG:Release>:>")
+
+target_compile_options(isal_asm PUBLIC "-I${QPL_SRC_DIR}/isal/include/"
+                       PUBLIC "-I${QPL_SRC_DIR}/isal/igzip/"
+                       PUBLIC "-I${QPL_SRC_DIR}/isal/crc/"
+                       PUBLIC "-DHAVE_AS_KNOWS_AVX512"
+                       PUBLIC "-DAS_FEATURE_LEVEL=10"
+                       PUBLIC "-DQPL_LIB")
+
+if (SANITIZE STREQUAL "undefined")
```

Comment:
For nasm target, we have to remove "-fno-sanitize=undefined" from COMPILE_OPTIONS, otherwise nasm compiler would fail to proceed due to unrecognition of "-fno-sanitize=undefined"

Reply:
Okay, thanks. Would you like to add that as a comment to the source code - it's not exactly obvious.

Reply:
Done